modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/ScriptPropertiesGeneratorTests.cs

```diff
@@ -0,0 +1,132 @@
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Godot.SourceGenerators.Tests;
+
+public class ScriptPropertiesGeneratorTestsTest : BaseTest<ScriptPropertiesGenerator>
+{
+    private readonly ITestOutputHelper _output;
+
+    public ScriptPropertiesGeneratorTestsTest(ITestOutputHelper output)
+    {
+        _output = output;
+
+        System.Console.SetOut(new OutputWriter(_output));
+    }
+
+    [Fact]
+    public async void EmptyTest()
+    {
+        var code = """
+        public partial class MyNode : Godot.Node
+        {
+        }
+        """;
+        var generated = """
+        using Godot;
+        using Godot.NativeInterop;
+
+        partial class MyNode
+        {
+        #pragma warning disable CS0109 // Disable warning about redundant 'new' keyword
+            /// <summary>
+            /// Cached StringNames for the properties and fields contained in this class, for fast lookup.
+            /// </summary>
+            public new class PropertyName : global::Godot.Node.PropertyName {
+            }
+        }
+
+        """;
+
+        await new CSharpSourceGeneratorVerifier<ScriptPropertiesGenerator>.Test
```

Comment:
The usual pattern, at least from what I've seen, is to have a static method in the verifier class, and then add an alias at the top of this file.

```csharp
using Verify = CSharpSourceGeneratorVerifier<ScriptPropertiesGenerator>;

await Verify.VerifySourceGeneratorAsync(source, generatedSources);
```

The static methods could be something like this:

```csharp
public static Task VerifySourceGeneratorAsync(string source, params (string FileName, string Source)[] generatedSources)
{
    var test = new Test()
    {
        TestState =
        {
            Source = { source },
            GeneratedSources = generatedSources
                .Select(generated => (GeneratedFileName(generated.FileName), SourceText.From(generated.Source, Encoding.UTF8)),
        },
    };
    return test.RunAsync();
}
```

This way we don't have to repeat the `SourceText.From` in every test. The `GeneratedFileName` method could also be moved to this class.